src/cairoUtilFuncGen/abi/abiEncode.ts

```diff
@@ -438,34 +494,60 @@ export class AbiEncode extends AbiBase {
       `}`,
     ].join('\n');
 
-    this.requireImport('warplib.maths.utils', 'felt_to_uint256');
-    this.auxiliarGeneratedFunctions.set(key, { name, code });
-    return name;
+    const genFuncInfo = {
+      name,
+      code,
+      functionsCalled: [
+        this.requireImport('warplib.maths.utils', 'felt_to_uint256'),
+        inlineEncoding,
+        valueEncoding,
+      ],
+    };
+    const auxFunc = this.createAuxiliarGeneratedFunction(genFuncInfo);
+
+    this.auxiliarGeneratedFunctions.set(key, auxFunc);
+    return auxFunc;
   }
 
-  private createStructInlineEncoding(type: UserDefinedType, def: StructDefinition) {
+  private createStructInlineEncoding(
+    type: UserDefinedType,
+    def: StructDefinition,
+  ): CairoFunctionDefinition {
     const key = 'struct inline ' + type.pp();
     const existing = this.auxiliarGeneratedFunctions.get(key);
-    if (existing !== undefined) return existing.name;
-
-    const instructions = def.vMembers.map((member, index) => {
-      const type = generalizeType(safeGetNodeType(member, this.ast.inference))[0];
-      const elemWidth = CairoType.fromSol(type, this.ast).width;
-      const readFunc = this.readMemory(type, 'mem_ptr');
-      const encoding = this.generateEncodingCode(
-        type,
-        'bytes_index',
-        'bytes_offset',
-        'element_offset',
-        `elem${index}`,
-      );
-      return [
-        `// Encoding member ${member.name}`,
-        `let (elem${index}) = ${readFunc};`,
-        `${encoding}`,
-        `let mem_ptr = mem_ptr + ${elemWidth};`,
-      ].join('\n');
-    });
+    if (existing !== undefined) return existing;
+
+    const decodingInfo: [string, CairoFunctionDefinition[]][] = def.vMembers.map(
+      (member, index) => {
+        const type = generalizeType(safeGetNodeType(member, this.ast.inference))[0];
```

Comment:
Destructuring

Reply:
I think there are other 100 lines similar to these where we just take the zero index...

Reply:
Yeah I've seen that there is a lot of the same code in other func generators. I've stopped commenting those because of that. So what do you propose? Focus on functionality not the code style while reviewing the PR? I don't think we've got the time to change the architecture here too so I assume we should just validate the correctness of the code? Maybe point out the bigger issues only?

Reply:
Because we're falling into the trap: fundamental changes are too big to fix in the PR and code style changes on the other hand are too numerous to fix. So we have to define what kind of issues we want to be fixed before merging the PR

Reply:
I believe is best to focus on correctness and fix style issues as we update the generators one by one for cairo1